go/types/rolling_value_hasher.go

```diff
@@ -5,18 +5,31 @@
 package types
 
 import (
-	"encoding/binary"
 	"sync"
 
-	"github.com/attic-labs/noms/go/hash"
+	"github.com/attic-labs/noms/go/sloppy"
+
 	"github.com/kch42/buzhash"
 )
 
 const (
 	defaultChunkPattern = uint32(1<<12 - 1) // Avg Chunk Size of 4k
 
-	// The window size to use for computing the rolling hash. This is way more than necessary assuming random data (two bytes would be sufficient with a target chunk size of 4k). The benefit of a larger window is it allows for better distribution on input with lower entropy. At a target chunk size of 4k, any given byte changing has roughly a 1.5% chance of affecting an existing boundary, which seems like an acceptable trade-off.
-	defaultChunkWindow = uint32(64)
+	// The choice of hash window here is a trade off between:
+	//   -When mutating a prolly tree, the likelihood that a change within a node
+	//    "cascades" by affecting the chunk boundary and causing a following node
+	//    to change.
+	//   -The chance that repeated sequences of bytes which happen to be longer
+	//    than the window do not chunk and cause average chunk size to be way
+	//    larger than the target
+	// The most likely source of repeated sequences larger than the chunk window
+	// is structs, all of whose field names are encoded together. A window size
+	// of 256 gives a roughly 6% chance that a change within a chunk will cascade
+	// into the next chunk while being sufficiently large to be larger than the
+	// field encodings of a "reasonable" struct.
+	defaultChunkWindow = uint32(256)
+
+	defaultCompressionWindow = uint16(1<<12) - 1 // 4k
```

Comment:
Why 4k?

Recall that sloppy prefers the most *recent* possible copy. Consider two cases: a map leaf containing a bunch of small structs: 

S1, S2, S3, S4, S5, S6.... 

and one containing one or two large structs

L1, (L2).

In the first case, the choice of 4k isn't likely to have a significant impact, because each struct will tend to refer to bytes in the struct before it (common field names). Obviously, it's possible to refer farther back, but more likely because of random common "content".

In the second case, the choice of 4k means that there's a much great chance that common content in L2 can refer to bytes in L1 and thus "fit" into the chunk. In practice, this means that larger value stores in maps/sets will still benefit from compression (greater density within the chunk).